### PR TITLE
Extended import via copy-paste for JSON and updated UI

### PR DESCRIPTION
**Pull request by IamEzio**
Fixes part of #2895 

This PR does the following:
1. Extended the import data via copy-paste functionality for JSON data.
2. Updated UI to allow users to import JSON files.
3. Added a link in Import datafile page to the documentation ([Link](https://github.com/centerofci/mathesar/blob/develop/docs/docs/user-guide/importing-data.md)).

**Screenshots**

https://github.com/centerofci/mathesar/assets/89375125/d9060a63-3b24-4227-9552-9a07d42c7ec2

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by IamEzio** [comment]
@dmos62 PTAL. Thanks!

**Comment by IamEzio** [comment]
@dmos62 PTAL. Thanks!

**Comment by IamEzio** [comment]
I am currently pointing to https://github.com/centerofci/mathesar/blob/develop/docs/docs/user-guide/importing-data.md for importing files documentation. I am waiting for changes to show on the site https://docs.mathesar.org/user-guide/users/, so I can point to that link.

File: mathesar_ui/src/pages/import-upload/UploadViaClipboard.svelte [hunk 1]
```diff
@@ -33,7 +33,10 @@
   <TextArea bind:value={clipboardContent} rows={10} disabled={isLoading} />
 </LabeledInput>
 <div class="help-content">
-  The data must be in tabular format (CSV, TSV etc.)
+  The data must be in tabular format (CSV, TSV etc) or a valid JSON. View <a
```
Comment (dmos62) [inline mathesar_ui/src/pages/import-upload/UploadViaClipboard.svelte:36]:
```suggestion
  The data must be in tabular format (CSV, TSV etc) or JSON. See relevant <a
```

File: mathesar_ui/src/pages/import-upload/UploadViaUrl.svelte [hunk 2]
```diff
@@ -37,7 +37,10 @@
 </LabeledInput>
 
 <div class="help-content">
-  The data must be in tabular format (CSV, TSV etc.)
+  The data must be in tabular format (CSV, TSV etc) or a valid JSON. View <a
+    href="https://github.com/centerofci/mathesar/blob/develop/docs/docs/user-guide/importing-data.md"
+    >details</a
+  >.
```
Comment (dmos62) [inline mathesar_ui/src/pages/import-upload/UploadViaUrl.svelte:43]: Could you replicate those changes to phrasing I suggested above?

**Comment by IamEzio** [comment]
Thanks @dmos62, I have made the changes. I have raised the PR #3023 to fix that doc navigation issue. Can you look into that first so I can use the correct link in this PR. Thanks!

**Comment by IamEzio** [comment]
@dmos62 I have updated the link to `https://docs.mathesar.org/user-guide/importing-data/` as per our offline discussion. PTAL. Thanks!

**Review by dmos62 — APPROVED** [review]

**Pull request merged**